src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -210,7 +220,24 @@ private void runRepair() {
         LOG.info("Repair for segment {} started, status wait will timeout in {} millis", segmentId,
             timeout);
         try {
-          condition.await(timeout, TimeUnit.MILLISECONDS);
+          long startTime = System.currentTimeMillis();
+          long maxTime = startTime + timeoutMillis;
+          
+          // If timeout is lower than 1mn, use timeout, otherwise we'll loop every minute to renew lead on segment
+          long waitTime = timeoutMillis<60000?timeoutMillis:60000;
+          
+          long lastLoopTime = System.currentTimeMillis();
+          while (System.currentTimeMillis() < maxTime) {
+            condition.await(waitTime, TimeUnit.MILLISECONDS);
+            if(lastLoopTime + 60_000 > System.currentTimeMillis() || context.storage.getRepairSegment(segmentId).get().getState() == RepairSegment.State.DONE){
+              // The condition has been interrupted, meaning the repair might be over
+              break;
+            }
+            
+            // Repair is still running, we'll renew lead on the segment when using Cassandra as storage backend
+            context.storage.renewLeadOnSegment(segmentId);
```

Comment:
If we're taken over 180 seconds to come back to this, (or any other partition failure within the process), then another reaper node could have taken this, but this code isn't dealing with that.